test/config.js

```diff
@@ -193,6 +176,114 @@ test('valid serviceName => active', function (t) {
   t.end()
 })
 
+test('serviceName defaults to package name', function (t) {
+  // TODO:
+  // - Move up top with other requires?
+  // - Switch to some promisified fs module?
+  var mkdirp = util.promisify(require('mkdirp'))
+  var rimraf = util.promisify(require('rimraf'))
+  var writeFile = util.promisify(fs.writeFile)
+  var symlink = util.promisify(fs.symlink)
+  var exec = util.promisify(cp.exec)
+
+  function testServiceConfig (pkg, handle) {
+    var tmp = path.join(os.tmpdir(), 'elastic-apm-node-test')
+    var files = [
+      {
+        action: 'mkdirp',
+        dir: tmp
+      },
+      {
+        action: 'create',
+        path: path.join(tmp, 'package.json'),
+        contents: JSON.stringify(pkg)
+      },
+      {
+        action: 'create',
+        path: path.join(tmp, 'index.js'),
+        contents: `
+          var apm = require('elastic-apm-node').start()
+          console.log(JSON.stringify(apm._conf))
+        `
+      },
+      {
+        action: 'mkdirp',
+        dir: path.join(tmp, 'node_modules')
+      },
+      {
+        action: 'symlink',
+        from: path.resolve(__dirname, '..'),
+        to: path.join(tmp, 'node_modules/elastic-apm-node')
+      }
+    ]
+
+    // NOTE: Reduce the sequence to a promise chain rather
+    // than using Promise.all(), as the tasks are dependent.
+    let promise = files.reduce((p, file) => {
+      return p.then(() => {
+        switch (file.action) {
+          case 'create': {
```

Comment:
TIL: I don't think I've ever seen blocks used together with `case` 😄It's not technically needed here, but I guess it does no harm

Reply:
Yeah, I do that all the time. It reads better to me, and scopes stuff better.